### DISCUSSION TITLE
I wonder why `Insn::Transaction` is added to every SELECT statement

### DISCUSSION THREAD

**Discussion by seonWKim** [tursodatabase/turso#292] [Q&A] [answered]
In `select.rs`'s `translate_select` function(line 432), `Insn::Transaction` is always added. But when I see the bytecodes generated by `sqlite3`, is doesn't always add `Transaction` instruction. What was the design choice for adding `Insn::Transaction` at the end of the select statement? 

![image](https://github.com/user-attachments/assets/bac881ac-8abb-457f-ae08-4e8bdc0626ed)
![Uploading image.png…]()

**Comment by penberg** [comment, marked as answer]
@seonWKim No particular reason except for thinking that SQLite also did that unconditionally. Let's fix it up!

**Reply by pereman2** [reply]
It looks like transactions is being skipped because you are not accessing any table and insead, using a scalar as a result. Meaning that transaction is not needed.
